### ISSUE TITLE
`TCopula`: add `:itau` (`ρ = sinpi(τ̂/2)`, `ν` by profile likelihood at fixed `ρ`) beside `:itau_irho`, and extend the correlation half to `d > 2`

### ISSUE THREAD

**Issue by dcelisgarza** [lrnv/Copulas.jl#502]
Written against `main` at 67dea70 (the 1.0 branch).

Companion to my Gaussian rank-inversion issue #501.

## Today

`TCopula` advertises `(:mle, :itau_irho)` in `d = 2` and `(:mle,)` above (`src/EllipticalCopulas/TCopula.jl:373`). `:itau_irho` (`:341`) takes the closed form `ρ = sinpi(τ̂/2)`, then identifies `ν` from Spearman's ρ by bisection on `|ρ_S(ν; ρ)| = |ρ̂_S|`. Two gaps for a downstream vine fitter:

1. **No `d > 2`.** The correlation half, `R = sinpi.(τ̂ ./ 2)` pairwise plus a correlation repair, is exactly as closed form in `d > 2` as in `d = 2`; only the `ν` half needs a one-dimensional search, which the `:mle` profile (`:305`) already performs.
2. **The estimator that vine libraries call `itau`** for the Student-t is `ρ` from τ and **`ν` from the likelihood with `ρ` fixed**, a one-dimensional profile. Your `:mle` is that profile over `λ = 1/ν` with `ρ` re-estimated at every `ν` through `_fit_t_corr_given_nu` (`:264`); `:itau` is the same loop with the correlation held at the τ inversion, so the inner optimisation disappears. A downstream selector that asks Copulas.jl for `:itau` per family such as VineCopulas.jl's `pair_method = :itau`, currently gets an `ArgumentError` for the Student-t and silently drops it from the candidate set.

## What I measured with some clanker help on Copulas 0.1.43, `d = 2`, 50 seeds each

I expected `:itau_irho` to identify `ν` poorly from the τ/ρ_S gap, so I had a look. It turns out the two rank estimates are strongly correlated on the same sample, so their gap is much less noisy than either method's estimate. Quantiles (5 %, 50 %, 95 %) of `ν̂`, and mean time per fit:

| truth | `n` | profile `:itau` | `:itau_irho` |
|---|---|---|---|
| `ν = 4, ρ = 0.6` | 1000 | 3.08 / 3.83 / 4.92, 0.08 s | 2.89 / 3.97 / 4.88, 0.39 s |
| `ν = 4, ρ = 0.6` | 250 | 2.56 / 3.95 / 10.6, 0.02 s | 2.11 / 3.60 / 6.31 (one `Inf`), 0.30 s |
| `ν = 8, ρ = 0.3` | 1000 | 4.92 / 6.96 / 12.6, 0.08 s | 4.95 / 6.90 / 11.8, 0.98 s |
| `ν = 4, ρ = 0.6` | 5000 | 3.57 / 3.88 / 4.78, 0.37 s | 3.41 / 3.88 / 4.89, 0.20 s |

So in `d = 2` the two are comparable in accuracy; the profile is faster for `n ≤ 1000` and slower at `n = 5000` (the quantile evaluation dominates), and it never returns `Inf` unless the Gaussian endpoint wins. The case for implementing `:itau` is therefore the difference in evaluation speed more than anything, that and it's my view that having alternate methods is nice to have, especially if it doesn't bring undue maintenance burden. If you would rather keep one method, extending `:itau_irho` to `d > 2` and registering `:itau` as an alias for it would also probably work.

## Proposal

```julia
function _fit(::Type{<:TCopula}, U, ::Val{:itau}; ν_bounds = (1.0, 100.0))
    R = _nearest_correlation(sinpi.(StatsBase.corkendall(U') ./ 2))   # helper from the Gaussian issue
    L = LinearAlgebra.cholesky(LinearAlgebra.Symmetric(R)).L
    lo, hi = ν_bounds
    loss = λ -> begin                                  # profile over λ = 1/ν, as :mle does
        iszero(λ) && return -Distributions.loglikelihood(GaussianCopula(R), U)
        ν = inv(λ)
        Z = Distributions.quantile.(Distributions.TDist(ν), U)
        return -_t_copula_loglik_factor(ν, L, Z)
    end
    res = Optim.optimize(loss, inv(hi), inv(lo), Optim.Brent())
    ν̂ = loss(0.0) <= Optim.minimum(res) ? Inf : inv(Optim.minimizer(res))
    return TCopula(ν̂, R)
end
_available_fitting_methods(::Type{<:TCopula}, d) = d == 2 ? (:mle, :itau, :itau_irho) : (:mle, :itau)
```

`_t_copula_loglik_factor(ν, L, Z)` is the existing helper (`:255`). `ν_bounds` follows the convention of the other keyword-controlled fitters, and a caller with its own box (a vine selector) narrows it. `:itau_irho` stays as it is.

## Tests

- `d = 2`, sample from `TCopula(4, [1 0.6; 0.6 1])`, `n = 5000`: `Σ[1,2] == sinpi(corkendall(U')[1,2]/2)` exactly and `ν̂ ∈ [3, 6]` in at least 95 of 100 seeds.
- On a Gaussian sample the profile returns `ν̂ = Inf` rather than the upper bound.
- `d = 3` on a known `R`: pairwise τ inversion to `1e-15`, `ν̂` within `[3, 6]`.
- `fit(CopulaModel, TCopula, U; method = :itau)` reports the method and the same `ν̂`.
- The 50-seed table above as a benchmark row, not an assertion.

Happy to send this and the Gaussian one as a single PR.

**Comment by lrnv** [comment]
Please send a PR. As you can see, we are making a lot of noise toward a 1.0 version, so much changes since 0.1.42 (last release) that i'm starting to be less and less confortable about the situation. But its necessary, our API promises were so bad. 

Its also the right time for fixing stuff, so i'll hapilly review your PR.

**Comment by dcelisgarza** [comment]
Don't worry, this is how it is. You truly can't know what the best API is until you start making and using the damn thing! I'll make the PR with both things simultaneously/

**Comment by lrnv** [comment]
Perfect. If you also want to take a shot at other 1.0-tagged issues, be my guest 🤣

**Comment by dcelisgarza** [comment]
Fixed in #507
